egui_demo_lib/src/apps/demo/window_with_panels.rs

```diff
@@ -0,0 +1,89 @@
+use egui::{menu, Align, CentralPanel, Layout, ScrollArea, SidePanel, TopBottomPanel};
+
+#[derive(Clone, PartialEq, Default)]
+#[cfg_attr(feature = "persistence", derive(serde::Deserialize, serde::Serialize))]
+pub struct WindowWithPanels {}
+
+impl super::Demo for WindowWithPanels {
+    fn name(&self) -> &'static str {
+        "🗖 Window With Panels"
+    }
+    fn show(&mut self, ctx: &egui::CtxRef, open: &mut bool) {
+        use super::View;
+        let window = egui::Window::new("Window with Panels")
+            .scroll(false)
+            .title_bar(true)
+            .resizable(true)
+            .collapsible(false)
+            .enabled(true)
+            .open(open);
+        window.show(ctx, |ui| self.ui(ui));
+    }
+}
+
+impl super::View for WindowWithPanels {
+    fn ui(&mut self, ui: &mut egui::Ui) {
+        let left_panel_min_width = 100.;
+        let left_panel_max_width = left_panel_min_width * 4.;
+        let bottom_height = 25.;
+
+        ui.expand_to_include_rect(ui.max_rect()); // Expand frame to include it all
+
+        let mut top_rect = ui.available_rect_before_wrap_finite();
+        top_rect.min.y += ui.spacing().item_spacing.y;
+        let mut top_ui = ui.child_ui(top_rect, Layout::top_down(Align::Max));
```

Comment:
Ideally we should have padding options on the Panel directly so we don't need to do this.